utils/http.go

```diff
@@ -163,3 +166,22 @@ func BuildURL(r *rest.Request, template string, params map[string]string) *url.U
 
 	return url
 }
+
+// ApiErr wraps errors returned by our APIs
+type ApiErr struct {
+	Error string `json:"error"`
+}
+
+func ParseApiErr(r *http.Response) (*ApiErr, error) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var aerr ApiErr
+	if err = json.Unmarshal(body, &aerr); err != nil {
+		return nil, err
+	}
+
+	return &aerr, nil
+}
```

Comment:
It would be better if ApiError was defined in `api/errors` package instead of `utils`.

Reply:
all of the ApiError fixes here https://github.com/mendersoftware/go-lib-micro/pull/37